### PR TITLE
use MethodName.File when value ends with .py

### LINKED ISSUE #5289: [CT-680] [Feature] Support for .py files when using file selectors
### Is this your first time opening an issue?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)

### Describe the Feature

> dbt should just know that when i say: dbt run -m my_model.sql i mean: dbt run -m my_model
> 
> https://twitter.com/pedram_navid/status/1519160872588898304

but for `.py` files, since [dbt Python models are gonna happen](https://github.com/dbt-labs/dbt-core/discussions/5261).

I would edit [this code](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/graph/selector_spec.py#L83), as well as (I think?) [this test](https://github.com/jwills/dbt-core/blob/1510eeb06a4479f5e74cac645f3b19d10fecaf85/test/unit/test_graph_selector_methods.py#L708).

### Describe alternatives you've considered

Life would go on without this feature, but why should we settle for less when a first-time contributor (me) wants to build it.

### Who will this benefit?

- > Pedram, apparently
- Me, because I'll gain experience by making a small contribution to `dbt-core`.
- The community, because with my newly gained experience, I'll be able to help with more, and more complex, issues over time.
- There are times when it's easier to include the file extension in the `dbt run` command. For example, if you're generating the `dbt run` command from a script, and don't want to bother stripping the file extensions from the list of models you plan to run.

### Are you interested in contributing this feature?

I'd like to create the PR, but i'll take any help / suggestions / guidance

### Anything else?

- [Josh's issue](https://github.com/dbt-labs/dbt-core/issues/5240) adding file selectors and `.sql` support.
- [Josh's PR](https://github.com/dbt-labs/dbt-core/pull/5241) solving that issue
- Wherein @jwills [suggested](https://github.com/dbt-labs/dbt-core/discussions/5261#discussioncomment-2771140), albeit with a winky face, I add support for `.py` files.

### PR DESCRIPTION
**Pull request by leoebfolsom**
Resolves #5289

### Description

This allows the user to write a command such as `dbt run -m my_fancy_stats_model.py`. Currently, a model ending in `.py` (acknowledging such models don't yet exist) cannot include the file extension in the `dbt run -m` command.

My approach here was to create a new fixture called `table_model_py`. This allowed me to not touch `table_model` but still achieve the desired functionality. I thought this was a reasonable first step, in the spirit of "optimize later." 

However, I think a cleaner approach would be either:
a) Updating `table_model` to allow for both `.sql` and `.py` extensions; or
b) Rename `table_model` to be `table_model_sql`, and maintain it alongside `table_model_py`.

I'm guessing `a` would be preferred (more scalable, less repeated code and less code in general), but I didn't want to build something like that out without getting input from others.

This is my first (attempted) contribution to dbt-core, so I may be doing things wrong--comments/guidance are welcome and desired! Thank you.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Comment by ChenyuLInx** [comment]
@leoebfolsom  the changes and test look great to me!

We are going to hold off on merging this until the python model is put out into the main branch.

File: core/dbt/graph/selector_spec.py
Comment (jtcohen6) [inline core/dbt/graph/selector_spec.py:83]: (Out of scope for this PR, but: should we also have `".csv"` here?)

Comment (ChenyuLInx) [reply core/dbt/graph/selector_spec.py:83]: is this for `dbt seed`?

Comment (jtcohen6) [reply core/dbt/graph/selector_spec.py:83]: yup

Comment (leoebfolsom) [reply core/dbt/graph/selector_spec.py:83]: I'll defer to @jtcohen6 and @ChenyuLInx on this decision. I'd have to review the dependencies / code in more detail to understand whether including `.csv` would require additional changes, and I'm sure you have a better sense for that. If it's simply a matter of adding the `.csv` to this one line to slightly expand this PR, I definitely don't object! ty

**Review by iknox-fa — APPROVED** [review]
This looks good to me.

**Comment by jtcohen6** [comment]
I think we're finally ready for this one :) @ChenyuLInx

**Comment by ChenyuLInx** [comment]
> I think we're finally ready for this one :) @ChenyuLInx

Haha I was looking for this today then saw your comment

File: core/dbt/graph/selector_spec.py
Comment (ChenyuLInx) [reply core/dbt/graph/selector_spec.py]: #5578 for this comment

File: tests/adapter/dbt/tests/adapter/basic/test_base.py
Comment (ChenyuLInx) [inline tests/adapter/dbt/tests/adapter/basic/test_base.py:25]: I think we probably don't want to add this to the base adapter tests as not all adapters support running this

Comment (ChenyuLInx) [reply tests/adapter/dbt/tests/adapter/basic/test_base.py:25]: Just the unittest above looks great to me

**Review by ChenyuLInx — CHANGES_REQUESTED** [review]

**Comment by ChenyuLInx** [comment]
@leoebfolsom  we are back since PR for python model just merged. I am going to just remove the changes in integration test and just push this in. Let me know if there's anything else you would want us to update!

**Review by ChenyuLInx — APPROVED** [review]

File: tests/adapter/dbt/tests/adapter/basic/test_base.py
Comment (leoebfolsom) [reply tests/adapter/dbt/tests/adapter/basic/test_base.py]: All right, and it looks like you made that change, so you don't need anything from me?

Comment (leoebfolsom) [reply tests/adapter/dbt/tests/adapter/basic/test_base.py]: Just saw your other comment--looks good to me! Thank you!

Comment (ChenyuLInx) [reply tests/adapter/dbt/tests/adapter/basic/test_base.py]: Sorry I just went ahead and did it thinking that it would be great for this to be included in the Beta release coming up soon!

Thank you so much for contributing this and the super quick response!!!

**Comment by ChenyuLInx** [comment]
Close and reopen to rerun synk check

**Pull request merged**
